runtime/src/lib.rs

```diff
@@ -106,22 +106,17 @@ macro_rules! step {
 /// EVM runtime.
 ///
 /// The runtime wraps an EVM `Machine` with support of return data and context.
-pub struct Runtime<'config> {
+pub struct Runtime {
 	machine: Machine,
 	status: Result<(), ExitReason>,
 	return_data_buffer: Vec<u8>,
 	context: Context,
-	_config: &'config Config,
+	_config: Config,
 }
 
-impl<'config> Runtime<'config> {
+impl Runtime {
 	/// Create a new runtime with given code and data.
-	pub fn new(
```

Comment:
I think we better not break the public APIs? What if we keep accepting a reference to config, but just clone it inside the new method?

I guess you won't need the 'config lifetime in this case too, as the reference will only live long enough to be cloned.